ssl-config/src/main/java/com/palantir/remoting/ssl/SslSocketFactories.java

```diff
@@ -0,0 +1,142 @@
+/*
+ * Copyright 2015 Palantir Technologies, Inc. All rights reserved.
+ */
+
+package com.palantir.remoting.ssl;
+
+import com.google.common.base.Throwables;
+import java.io.IOException;
+import java.io.InputStream;
+import java.nio.file.Files;
+import java.security.GeneralSecurityException;
+import java.security.Key;
+import java.security.KeyStore;
+import java.security.cert.Certificate;
+import javax.net.ssl.KeyManager;
+import javax.net.ssl.KeyManagerFactory;
+import javax.net.ssl.SSLContext;
+import javax.net.ssl.SSLSocketFactory;
+import javax.net.ssl.TrustManager;
+import javax.net.ssl.TrustManagerFactory;
+
+/**
+ * Utility functions for creating {@link SSLSocketFactory}s
+ * that are configured with Java trust stores and key stores.
+ */
+public final class SslSocketFactories {
+    private SslSocketFactories() {}
+
+    /**
+     * Create an {@link SSLSocketFactory} from the provided configuration.
+     *
+     * @param config an {@link SslConfiguration} describing the trust store and key store configuration
+     * @return an {@link SSLSocketFactory} according to the input configuration
+     */
+    public static SSLSocketFactory createSslSocketFactory(SslConfiguration config) {
+        SSLContext sslContext = createSslContext(config);
+        return sslContext.getSocketFactory();
+    }
+
+    /**
+     * Create an {@link SSLContext} initialized from the provided configuration.
+     *
+     * @param config an {@link SslConfiguration} describing the trust store and key store configuration
+     * @return an {@link SSLContext} according to the input configuration
+     */
+    public static SSLContext createSslContext(SslConfiguration config) {
+        TrustManager[] trustManagers = createTrustManagerFactory(config.trust()).getTrustManagers();
+
+        KeyManager[] keyManagers = null;
+        if (config.key().isPresent()) {
+            keyManagers = createKeyManagerFactory(config.key().get()).getKeyManagers();
+        }
+
+        try {
+            SSLContext sslContext = SSLContext.getInstance("TLS");
+            sslContext.init(keyManagers, trustManagers, null);
+            return sslContext;
+        } catch (GeneralSecurityException e) {
+            throw Throwables.propagate(e);
+        }
+    }
+
+    private static TrustManagerFactory createTrustManagerFactory(TrustStoreConfiguration trustStoreConfig) {
+        try {
+            TrustManagerFactory trustManagerFactory = TrustManagerFactory.getInstance(
+                    TrustManagerFactory.getDefaultAlgorithm());
+            KeyStore keyStore = KeyStore.getInstance(trustStoreConfig.type().or("JKS"));
```

Comment:
Sure, that should be possible -- can make that change (was matching style/approach of what was previously done)